packages/react-jsx-babel-plugin/__tests__/__snapshots__/transform-jsx-to-react-jsx-test.js.snap

```diff
@@ -0,0 +1,424 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`transform react to jsx React.fragment to set keys and source 1`] = `
+"var _jsxFileName = \\"\\";
+var x = React.jsxDEV(React.Fragment, {}, \\"foo\\", true, {
+  fileName: _jsxFileName,
+  lineNumber: 1
+}, this);"
+`;
+
+exports[`transform react to jsx adds appropriate new lines when using spread attribute 1`] = `
+"React.jsx(Component, Object.assign({}, props, {
+  sound: \\"moo\\"
+}));"
+`;
+
+exports[`transform react to jsx arrow functions 1`] = `
+"var foo = function () {
+  var _this = this;
+
+  return function () {
+    return React.jsx(_this, {});
+  };
+};
+
+var bar = function () {
+  var _this2 = this;
+
+  return function () {
+    return React.jsx(_this2.foo, {});
+  };
+};"
+`;
+
+exports[`transform react to jsx assignment 1`] = `
+"var div = React.jsx(Component, Object.assign({}, props, {
+  foo: \\"bar\\"
+}));"
+`;
+
+exports[`transform react to jsx concatenates adjacent string literals 1`] = `
+"var x = React.jsx(\\"div\\", {
+  children: [\\"foo\\", \\"bar\\", \\"baz\\", React.jsx(\\"div\\", {
+    children: \\"buz bang\\"
+  }), \\"qux\\", null, \\"quack\\"]
+});"
+`;
+
+exports[`transform react to jsx display name assignment expression 1`] = `
+"var Component;
+Component = React.createClass({
+  render: function render() {
+    return null;
+  }
+});"
+`;
+
+exports[`transform react to jsx display name export default 1`] = `
+"export default React.createClass({
+  render: function render() {
+    return null;
+  }
+});"
+`;
+
+exports[`transform react to jsx display name if missing 1`] = `
+"var Whateva = React.createClass({
+  displayName: \\"Whatever\\",
+  render: function render() {
+    return null;
+  }
+});
+var Bar = React.createClass({
+  \\"displayName\\": \\"Ba\\",
+  render: function render() {
+    return null;
+  }
+});"
+`;
+
+exports[`transform react to jsx display name object declaration 1`] = `
+"exports = {
+  Component: React.createClass({
+    render: function render() {
+      return null;
+    }
+  })
+};"
+`;
+
+exports[`transform react to jsx display name property assignment 1`] = `
+"exports.Component = React.createClass({
+  render: function render() {
+    return null;
+  }
+});"
+`;
+
+exports[`transform react to jsx display name variable declaration 1`] = `
+"var Component = React.createClass({
+  render: function render() {
+    return null;
+  }
+});"
+`;
+
+exports[`transform react to jsx fragment with no children 1`] = `"var x = React.jsx(React.Fragment, {});"`;
```

Comment:
It wouldn't make sense if expected that to increase in scope. Fragment syntax is probably the only one that has special syntax so that makes sense.

`<></>` should auto-import.

It should not extend to things with identifiers though.

However, for example `<Fragment />` and `<Profiler />` shouldn't auto-import.